### ISSUE TITLE
python-ruamel-yaml and metrics on RHEL 7.3

### ISSUE THREAD

**Issue by glamperi** [openshift/openshift-ansible#3302]
It looks like the update of the git repo in the last few days uses
python-ruamel-yaml for metrics.  This does not work on RHEL 7.3, lucky
I cloned the repo, the 6-7 day old cloned git repo works.

**Comment by Adamjmb** [comment]
This issue is blocking. Is there a work around for this issue?

**Comment by detiber** [comment]
@sdodson @kwoodson @tdawson fyi

**Comment by etsauer** [comment]
It looks like, even in upstream fedora/cent channels the package might be under a different name:

f25:
```
$ sudo dnf search python*yaml
Last metadata expiration check: 4:03:01 ago on Fri Feb 10 17:30:23 2017.
============= N/S Matched: python*yaml 
python3-PyYAML.x86_64 : YAML parser and emitter for Python
python2-grafyaml.noarch : Tools to make Grafana dashboards from templates
python3-grafyaml.noarch : Tools to make Grafana dashboards from templates
python2-ruamel-yaml.x86_64 : YAML 1.2 loader/dumper package for Python
```

rhel 7.3:
```
[root@master-0 ~]# yum repolist
Loaded plugins: search-disabled-repos
repo id                                                                                        repo name                                                                                                     status
!rhel-7-server-extras-rpms/x86_64                                                              Red Hat Enterprise Linux 7 Server - Extras (RPMs)                                                                393
!rhel-7-server-ose-3.4-rpms/x86_64                                                             Red Hat OpenShift Container Platform 3.4 (RPMs)                                                                  603
!rhel-7-server-rpms/7Server/x86_64                                                             Red Hat Enterprise Linux 7 Server (RPMs)                                                                      27,130
repolist: 28,126
[root@master-0 ~]# yum search *python*yaml*
Loaded plugins: search-disabled-repos
Warning: No matches found for: *python*yaml*
No matches found
```

**Comment by warmchang** [comment]
+1. I got the error about "python-ruamel-yaml" when upgrade the origin version.

> TASK [lib_openshift : lib_openshift ensure python-ruamel-yaml package is on target] ***
> fatal: [master.example.com]: FAILED! => {
>     "changed": false,
>     "failed": true,
>     "rc": 126,
>     "results": [
>         "No package matching 'python-ruamel-yaml' found available, installed or updated"
>     ]
> }
> 
> MSG:
> 
> No package matching 'python-ruamel-yaml' found available, installed or updated
> 
> PLAY RECAP *********************************************************************
> localhost                  : ok=30   changed=0    unreachable=0    failed=0
> master.example.com         : ok=293  changed=21   unreachable=0    failed=1


The host is CentOS 7.3:

> [root@master openshift-ansible]# cat /etc/redhat-release
> CentOS Linux release 7.3.1611 (Core)

**Comment by etsauer** [comment]
I was able to work around the issue with the following enhancement to lib_openshift:

```
$ git diff roles/lib_openshift/tasks/main.yml
diff --git a/roles/lib_openshift/tasks/main.yml b/roles/lib_openshift/tasks/main.yml
index 2980c8a..3778c9e 100644
--- a/roles/lib_openshift/tasks/main.yml
+++ b/roles/lib_openshift/tasks/main.yml
@@ -1,5 +1,15 @@
 ---
+- set_fact:
+    os_version: "{{ ansible_distribution_version.split('.') }}"
+
+- set_fact:
+    python_yaml_pkg: python-ruamel-yaml
+
+- set_fact:
+    python_yaml_pkg: PyYAML
+  when: ansible_distribution == "RedHat" and os_version[0] | int == 7 and os_version[1] | int >= 3
+
 - name: lib_openshift ensure python-ruamel-yaml package is on target
   package:
-    name: python-ruamel-yaml
+    name: "{{ python_yaml_pkg }}"
     state: present
```

however, other things within `playbooks/common/openshift-cluster/openshift_hosted.yml` around metrics seems to be broken.

**Comment by xelfe** [comment]
I have the same problem with CentOS Linux release 7.3.1611 (Core) when trying to update 3.3 to 3.4

**Comment by ollyhewett** [comment]
Same here. 
Built environments of Centos 7.3.1611 & Fedora 25 base builds and both fail with this error

**Comment by sdodson** [comment]
We're working to get this dependency into EPEL, Fedora 25, and the Centos PaaS SIG repo. The release-1.4 branch does not have this dependency and is the most stable branch for installing Origin 1.4 if that's your target.

**Comment by sdodson** [comment]
@etsauer That won't work as we start to actually use functionality of python-ruamel-yaml it will break down. I don't suggest that.

**Comment by smarterclayton** [comment]
https://ci.openshift.redhat.com/jenkins/job/zz_origin_gce_image/113/console failed to pick up latest because the ops-mirrors don't have it.

**Comment by warmchang** [comment]
@sdodson , update successfully user release-1.4 branch.

**Comment by cgwalters** [comment]
I'm not sure whether to overload this bug, but we need to choose how to handle this new dependency for Atomic Host.

**Comment by sdodson** [comment]
> I'm not sure whether to overload this bug, but we need to choose how to handle this new dependency for Atomic Host.

For now we've chosen to fall back to using PyYAML which is currently available on atomic host, right? https://github.com/openshift/openshift-ansible/pull/3383

**Comment by xiaoping378** [comment]
the issue is also met at master.

**Comment by php-coder** [comment]
The workaround that worked for me with Fedora 24:
```diff
--- roles/lib_openshift/tasks/main.yml
+++ roles/lib_openshift/tasks/main.yml
@@ -4,8 +4,8 @@
     path: /run/ostree-booted
   register: ostree_booted
 
-- name: lib_openshift ensure python-ruamel-yaml package is on target
-  package:
-    name: python2-ruamel-yaml
-    state: present
-  when: not ostree_booted.stat.exists
```

**Comment by sdodson** [comment]
This should be fixed now via #3499
